### PR TITLE
Scheduled post option bot time exclusion

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Hid user's timezone in options if DM user is a bot

#### Ticket Link
Fixes https://mattermost.atlassian.net/browse/MM-61859

#### Release Note
```release-note
none
```

### CODE REVIEW

**Review by larkox — APPROVED** [review]

**Comment by larkox** [comment]
@harshilsharma63 Does this also solve it for the modal when choosing a custom time?

File: webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx [hunk 1]
```diff
@@ -43,27 +50,27 @@ function useTimePostBoxIndicator(channelId: string) {
...
+
+        return getTimezoneForUserProfile(teammate);
+    }, [teammate]);
 
     // current user timezone
     const userCurrentTimezone = useSelector((state: GlobalState) => getCurrentTimezone(state));
 
     // UseEffect to update the timestamp and the visibility for the time indicator
     useEffect(() => {
+        if (isDM && teammate?.is_bot) {
```
Comment (larkox) [inline webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx:67]: Should we add the self dm check here too?

**Review by larkox — APPROVED** [review]
Just one check

File: webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx (hunk 1 above)
Comment (harshilsharma63) [reply webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx:67]: Its already there, maybe you had an old tab open?

Comment (larkox) [reply webapp/channels/src/components/advanced_text_editor/use_post_box_indicator.tsx:67]: No, in this file there is no self dm check. It is on the other file, though.

**Comment by NCPSNetworks** [comment]
Hello, there seems to be an issue with this pull request. Reproduction steps:

1. Open Developer Tools (F12), disable cache
2. Log in to Mattermost without being logged in previously
3. A white screen is displayed, and errors are shown in the develoepr console

I'm not sure if it's due to my configuration error. When I roll back the changes from this pull request, everything works fine.

Here are the error screenshots:
42559ed6cdd90633814ade25b7e9333f

**Comment by harshilsharma63** [comment]
@NCPSNetworks thanks for reporting this.

I tried this a lot of times could never reproduce this locally. However, someone else also reported the same even even when they are logged in. There is a null check missing in that file and maybe thats causing this error but I don't understand why why that code - belonging to a component that specifically renders when a channel is loaded and that channel is a DM, would cause an error for a non-logged in user.

I'll dive into this a bit and see if I can figure this out.

**Comment by NCPSNetworks** [comment]
@harshilsharma63 thank you for your reply.
Actually, this white screen error occurred right after I clicked the login button and successfully logged in, before any channels had a chance to load. After the white screen appears, I have to reload the webpage to return to normal.

Thank you!

**Comment by harshilsharma63** [comment]
@NCPSNetworks and are you seeing this error on a local Mattermost or on a server?

**Comment by NCPSNetworks** [comment]
@harshilsharma63 I encountered this issue on the server, which is built and deployed from the master branch.

When I individually remove the code from this pull request, I don't encounter the issue.

There are several users and a Copilot Bot in the server's DM channel.

**Comment by harshilsharma63** [comment]
Okay, I'll test this out. Thanks!

**Pull request merged**